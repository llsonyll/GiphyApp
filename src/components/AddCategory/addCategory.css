/* .addCategory {
    border: 1px solid blue;
} */