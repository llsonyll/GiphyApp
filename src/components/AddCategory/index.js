export { default } from "./addCategory.jsx";
