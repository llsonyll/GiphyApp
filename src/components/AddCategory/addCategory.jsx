import { useState } from "react";
import PropTypes from "prop-types";
import "./addCategory.css";

const AddCategory = ({ addNewCategory }) => {
  const [input, setInput] = useState("");

  const handleInputChange = (e) => {
    setInput(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (input.trim().length <= 1) return;
    addNewCategory(input);
    setInput("");
  };

  return (
    <div className="addCategory">
      <form onSubmit={handleSubmit} aria-label="form">
        <input
          type="text"
          value={input}
          onChange={handleInputChange}
          placeholder="Buscar gifs"
        />
      </form>
    </div>
  );
};

AddCategory.propTypes = {
  addNewCategory: PropTypes.func.isRequired,
};

export default AddCategory;
