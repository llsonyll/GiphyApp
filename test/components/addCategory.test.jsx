import { render, screen, fireEvent } from "@testing-library/react";

import AddCategory from "../../src/components/AddCategory";

describe("Add Category Test", () => {
  test("should receive a function as prop", () => {
    const { container } = render(<AddCategory addNewCategory={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  test("should read input change", () => {
    render(<AddCategory addNewCategory={() => {}} />);
    const inputValue = "Lolito";
    const input = screen.getByRole("textbox");
    fireEvent.input(input, {
      target: {
        value: inputValue,
      },
    });

    expect(input.value).toBe(inputValue);
  });

  test("should call addNewCategory if input got a value", () => {
    const inputValue = "Lolito";
    const addNewCategory = jest.fn();
    render(<AddCategory addNewCategory={addNewCategory} />);

    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    fireEvent.input(input, {
      target: {
        value: inputValue,
      },
    });

    fireEvent.submit(form);

    expect(input.value).toBe("");
    expect(addNewCategory).toHaveBeenCalled();
    expect(addNewCategory).toHaveBeenCalledTimes(1);
    expect(addNewCategory).toHaveBeenCalledWith(inputValue);
  });

  test("should not call addNewCategory if input got an empty value", () => {
    const inputValue = "";
    const addNewCategory = jest.fn();
    render(<AddCategory addNewCategory={addNewCategory} />);

    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    fireEvent.input(input, {
      target: {
        value: inputValue,
      },
    });

    fireEvent.submit(form);

    expect(input.value).toBe("");
    expect(addNewCategory).not.toHaveBeenCalled();
    expect(addNewCategory).toHaveBeenCalledTimes(0);
  });
});
