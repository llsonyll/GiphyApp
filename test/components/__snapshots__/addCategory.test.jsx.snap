// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Category Test should receive a function as prop 1`] = `
<div>
  <div
    class="addCategory"
  >
    <form
      aria-label="form"
    >
      <input
        placeholder="Buscar gifs"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
